core/common/src/main/scala/org/typelevel/otel4s/context/propagation/PassThroughPropagator.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.context
+package propagation
+
+import cats.effect.SyncIO
+import org.typelevel.otel4s.context.syntax._
+
+/** A [[TextMapPropagator]] that extracts a specified collection of fields and
+  * stores them in a context, and extracts them from a context later for
+  * injection. It does not interact with telemetry.
+  */
+final class PassThroughPropagator[Ctx, K[X] <: Key[X]] private (
+    val fields: Iterable[String],
```

Comment:
All constructor methods (e.g. `apply`) use `forDistrinctFields` under the hood. 

If the class is intended to work only with unique fields, should we use `Set[String]` instead?

Reply:
we never do any lookups from it, so it has no particular performance advantage that I can think of. and `Set` is not ordered, so to the minor extent that the user cares for the order of the fields as rendered by `toString`, `Seq` is very slightly nicer. *very* slightly. I don't feel strongly however

Reply:
> Set is not ordered

valid points against using it. I agree